packages/engine/src/authoring/AuthoringState.tsx

```diff
@@ -267,21 +274,31 @@ export const AuthoringState = defineState({
   }
 })
 
-const SourceReactor = (props: { entity: Entity; layer: LayerID }) => {
-  const loaded = GLTFComponent.useSceneLoaded(props.entity)
+const SourceReactor = (props: { entity: Entity }) => {
+  const authoringEntity = getAuthoringCounterpart(props.entity)
+
+  const hasScene = useHasComponent(props.entity, SceneComponent)
+  const isOwnedByScene = useOptionalComponent(props.entity, NetworkObjectComponent)?.ownerId.value == SceneUser
+  const isChildOfScene = !!useAncestorWithComponents(props.entity, [SceneComponent])
+
+  const valid = authoringEntity || hasScene || (isOwnedByScene && isChildOfScene)
+
+  const entity = authoringEntity || props.entity
+  const loaded = GLTFComponent.useSceneLoaded(entity)
 
   useEffect(() => {
-    if (!loaded) return
+    if (!loaded || !valid) return
 
-    const sourceID = UUIDComponent.getAsSourceID(props.entity)
-    const sourceData = getSourceSnapshot(sourceID, props.layer)
+    const layer = authoringEntity ? Layers.Authoring : Layers.Simulation
+    const sourceID = UUIDComponent.getAsSourceID(entity)
+    const sourceData = getSourceSnapshot(sourceID, layer)
 
     dispatchAction(AuthoringActions.initialize({ sourceID, partialState: sourceData }))
 
     return () => {
       dispatchAction(AuthoringActions.uninitialize({ sourceID }))
     }
-  }, [loaded])
+  }, [loaded, valid])
```

Comment:
do we need to merge this into a single bool dep to avoid it retriggering due to potential race conditions with the network object component?